### ISSUE TITLE
GUI often hangs in Win8.1

### ISSUE THREAD

**Issue by Ergonomicmike** [portapps/slack-portable#16]
I run Slack Portable in both Win7 64 HP and Win8.1 64 Pro. (Albeit in a Comodo virtualized sandbox in both.)

When running in Win7, the GUI works fine. But in Win8.1, the mouse can only switch one Channel. Then the mouse won't work anymore. Also, the Channel history won't fill beyond one screen. (That is, I can't scroll up.) And when I maximize the GUI, I get a blank white screen. (I can post a video if that will help show this better.)

#### Steps to reproduce this issue

1. Run Slack Portable in Win8.1.
2. Sign in to your Slack group.
3. Try to change Channels or scroll the Channel History.

FWIW, I have noticed that Windows Themes affect GUI's in Win8.1 more than they do in Win7. (For example, there's an adaptive setting for window title colors, etc.) In Palemoon browser, there is a setting to tell Palemoon to ignore High Contrast themes. (I am not running High Contrast. I just mention this particular behavior of Win8.1.)

**Comment by crazy-max** [comment]
Hi @Ergonomicmike, I cannot reproduce your issue on Windows 8.1 Pro 64-bits VM. It's quite difficult to reproduce this kind of problem and I don't think it has anything to do with making the application portable with Portapps.
Have you tried with the official installer? If you still have this issue, I therefore suggest that you look for a solution on [Slack's Help Center](https://get.slack.help/hc/en-us)

**Comment by Ergonomicmike** [comment]
Thanks for checking.

I haven't tried the official installer. That's the main reason why I sought out the Portable App. :-)

To be fair, I'm having this issue on a heavily NTLite'd processed Win8.1.

The issue comes and goes - that is, sometimes it runs fine in my 'Lite'd Win8.1.

The easiest thing for me to try next is to run it outside Comodo's sandbox.

**Comment by crazy-max** [comment]
Oh gosh, so much layers :dizzy_face:
If you can first use the official installer and try it for a while :slightly_smiling_face:

**Comment by Ergonomicmike** [comment]
Okay, I tried running Slack Portable outside Comodo's Sandbox in Win8.1.

It works okay that way.

It's only when I run it "virtualized" using Comodo that it freezes. (For example, I can click on File > Preferences, but the Preferences won't open.)

I'm thinking that this problem started for me in Win8.1 when I switched to the 3.3.3.40 version of Slack Portable.

The strange thing is that I'm also running the .40 version in Win7 (I point Comodo to the same folder in both) and Slack Portable runs fine there in Win7.

If I had to guess as a non-developer, I would guess that there's some process / thread that gets spawned for scrolling, caching, painting new windows - something like that - that Comodo is not allowing to run under Win8.1 but allows under Win7. (Different task manager/scheduler?)

I completely uninstalled, cleaned, and reinstalled Comdo in Win8.1. No change.

You might say "Well, just run Slack Portable without sandboxing it." 

I could. But when I run it in Comodo's sandbox, I get warnings about how Slack is always trying to "access memory." (Blocked by Comodo.)

I get a lot of warnings about that while Slack is running.

I don't know what memory Slack is trying to access. I don't get those warnings when I run Palemoon browser virtualized. 

I don't know if Slack is sniffing my computer or not. Since I don't know, I prefer to run it sandboxed.

I'll think about making an image of my Win8.1 and using Slack's official installer and running that virtualized. (I just hate having programs writing to my Registry. :-(  )

**Comment by Ergonomicmike** [comment]
Funny thing - while writing the above, I had Slack Portable virtual running max size Window. It started working correctly. (Could get to Preferences, etc.) 

Then I resized the Window and it hung. 

Caching something?

**Comment by crazy-max** [comment]
Ok but have you tried the [official Slack desktop setup](https://slack.com/downloads/windows) in your Comodo's sandbox ? It would rule out any problems with the portable version. If not, I will reopen this issue.

**Comment by Ergonomicmike** [comment]
I found the problem. It's caused by Slack calling a Windows exe (and related dll?) called RuntimeBroker.exe. 

From what I can tell from perusing the web, RuntimeBroker is a Metro/Store function that checks Apps for notifications/permissions. (?)

That explains why it wasn't a problem in Win7. (No Metro/Store stuff.)

Comdo virtualized that executable, and perhaps virtualizing it caused a race condition or a delay that would sometimes hang Slack.

I ended up hacking Windows and renamed the exe to exe.old.

So not a Slack Portable problem per se. Just something in Slack that manifests itself when sandboxed in Win8.1. (I presume would also be a problem in WIn10, since WIn10 also has RuntimeBroker.)
